packages/language-server/src/language.ts

```diff
@@ -54,11 +57,19 @@ export class Html1File implements VirtualFile {
 
 	addStyleTag() {
 		let i = 0;
+		this.cssTokens = [];
 		this.htmlDocument.roots.forEach(root => {
 			if (root.tag === 'style' && root.startTagEnd !== undefined && root.endTagStart !== undefined) {
 				const styleText = this.snapshot.getText(root.startTagEnd, root.endTagStart);
+				const fileName = this.fileName + `.${i++}.css`;
+
+				const document = html.TextDocument.create(fileName, 'css', 0, styleText);
+				const stylesheet = cssLs.parseStylesheet(document);
```

Comment:
If performance is important, you should defer parsing stylesheets until `provideCompletionItems` called.